src/mjolnir/shortcutbuilder.cc

```diff
@@ -452,12 +453,17 @@ uint32_t AddShortcutEdges(GraphReader& reader,
         // end node in the new level). Keep track of the last restriction
         // on the connected shortcut - need to set that so turn restrictions
         // off of shortcuts work properly
-        length += ConnectEdges(reader, end_node, next_edge_id, shape, end_node, opp_local_idx, rst,
-                               average_density, total_duration, total_truck_duration);
+        ConnectEdges(reader, end_node, next_edge_id, shape, end_node, opp_local_idx, rst,
+                     average_density, total_duration, total_truck_duration);
       }
+
       // Names can be different in the forward and backward direction
       bool diff_names = tilebuilder.OpposingEdgeInfoDiffers(tile, directededge);
 
+      // Get the length from the shape. This prevents roundoff issues when forming
+      // elevation.
+      uint32_t length = valhalla::midgard::length(shape);
```

Comment:
Length of shortcut is not always equal to the sum of the links of the directed edges comprising the shortcut due to directed edge length being cast to integer - thus leading to roundoff errors. This discrepancy between shortcut length (stored in the directed edge) and shape length can lead to elevation encoding issues.

Reply:
good catch!